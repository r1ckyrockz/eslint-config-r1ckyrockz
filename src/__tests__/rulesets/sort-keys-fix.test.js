const { createSortKeysFixRules } = require('../../rulesets/sort-keys-fix');

describe('createSortKeysFixRules', () => {
  test('matches snapshot', () => {
    const project = {};

    expect(createSortKeysFixRules(project)).toMatchSnapshot();
  });

  test('allows passing custom rules', () => {
    const rule = 'sort-keys-fix/sort-keys-fix';
    const level = 'off';

    const project = {
      rules: {
        [rule]: level,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    const defaultRuleset = createSortKeysFixRules({
      ...project,
      rules: {},
    });

    const result = createSortKeysFixRules(project);

    expect(result[rule]).toBe(level);
    expect(result[rule]).not.toBe(defaultRuleset[rule]);

    expect(result).toMatchSnapshot();
  });
});
