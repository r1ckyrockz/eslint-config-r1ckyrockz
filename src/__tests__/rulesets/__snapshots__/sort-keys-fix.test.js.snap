// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSortKeysFixRules allows passing custom rules 1`] = `
Object {
  "sort-keys-fix/sort-keys-fix": "off",
}
`;

exports[`createSortKeysFixRules matches snapshot 1`] = `
Object {
  "sort-keys-fix/sort-keys-fix": "warn",
}
`;
